checkov/common/output/record.py

```diff
@@ -79,6 +80,7 @@ def __init__(
         self.vulnerability_details = vulnerability_details  # Stores package vulnerability details
         self.connected_node = connected_node
         self.guideline: str | None = None
+        self.details: List[str] = details
```

Comment:
```suggestion
        self.details: List[str] = details or []
```
this will create an empty `list`, if `details` is `None`